### PR TITLE
feat: add repo_dispatch trigger and options to build vite

### PR DESCRIPTION
**Pull request by dominikg**
from branch, tag or commit (#4)

Added cac for better cli handling and some code formatting too.

to trigger repository_dispatch, we can use this action https://github.com/peter-evans/repository-dispatch

### CODE REVIEW

File: .github/workflows/ecosystem-ci.yml [hunk 1]
```diff
@@ -21,7 +37,9 @@ jobs:
           node-version: 16
       - run: npm i -g pnpm
       - run: pnpm i --frozen-lockfile
-      - run: node index.js --viteBuildOnly # checkout, install and build vite
+      - name: build branch
+        if: ${{github.event.inputs.}}
+        run: node ecosystem-ci.js build-vite --${{ github.event.inputs.refType || github.event.client_payload.refType || 'branch' }} ${{ github.event.inputs.ref || github.event.client_payload.ref || 'main' }}
```
Comment (dominikg) [inline .github/workflows/ecosystem-ci.yml:42]: not exactly sure if chaining inputs works this way. Is there a unified api for action inputs?

File: .github/workflows/ecosystem-ci-selected.yml [hunk 2]
```diff
@@ -9,7 +9,21 @@ env:
 on:
   workflow_dispatch:
     inputs:
-      test:
+      ref:
+        description: 'vite ref to use'
+        required: true
+        type: string
+        default: 'main'
+      refType:
```
Comment (patak-dev) [inline .github/workflows/ecosystem-ci-selected.yml:17]: Should `ref` and `refType` be inverted here? So the flow is to decide what kind of ref you use and then you give the id for it?

**Review by patak-dev — APPROVED** [review]

**Comment by patak-dev** [comment]
Awesome! I'll let you merge this one

**Comment by dominikg** [comment]
this is the wrong branch, sorry... sending the right one

**Pull request closed without merging**